blog/2023-10-10-weaviate-1-22-release/index.mdx

```diff
@@ -0,0 +1,253 @@
+                    "dataType": ["text"],
+                    "name": "child"
+                }
+            ],
+        }
+    ],
+}
+```
+
+An example object of this class may look like this:
+
+```json
+{
+    "parent": {"child": "I'm nested!"}
+}
+```
+
+Each nested property can have its own data type, including being an `object` itself. This allows you to create nested structures of arbitrary depth.
+
+<details>
+  <summary>Two-level depth example</summary>
+
+The following class definition shows a `DoublyNested` class with a `parent` property that is an `object` datatype, similarly to above. Here, hwoever, the `child` property is itself an `object` datatype, and has its own `nestedProperties` of `childname` and `grandchild`.
+
+```json
+{
+    "class": "DoublyNested",
+    "properties": [
+        {
+            "dataType": ["object"],
+            "name": "parent",
+            "nestedProperties": [
+                {
+                    "dataType": ["object"],
+                    "name": "child",
+                    "nestedProperties": [
+                        {
+                            "dataType": ["text"],
+                            "name": "childname",
+                        },
+                        {
+                            "dataType": ["text"],
+                            "name": "grandchild"
+                        },
+                    ],
+                },
+            ],
+        }
+    ],
+}
+```
+
+An example object of this class may look like this:
+
+```json
+{
+    "parent": {
+        "child": {
+            "childname": "Nested child",
+            "granchild": "I'm doubly nested!"
+        }
+    }
+},
+```
+
+</details>
+
+More practically, this type of structure may be applied to capture addresses, product information, or any other nested data structure. Take a look at the following example, which shows a `Person` class with an `address` property that is an `object` datatype. The `address` property is able to store its own `nestedProperties` of `street` and `city`.
+
+```json
+{
+    "class": "Person",
+    "properties": [
+        {
+            "dataType": ["text"],
+            "name": "last_name",
+        },
+        {
+            "dataType": ["object"],
+            "name": "address",
+            "nestedProperties": [
+                {"dataType": ["text"], "name": "street"},
+                {"dataType": ["text"], "name": "city"}
+            ],
+        }
+    ],
+}
+```
+
+The user may then create an object of this class, such as:
+
+```json
+{
+    "last_name": "Franklin",
+    "address": {
+        "city": "London",
+        "street": "King Street"
+    }
+}
+```
+
+### Auto-schema support for nested properties
+
+Given that the `object` data types allow nested properties of arbitrary depth, you will be pleased to know that auto-schema supports nested properties. This means that you can create a class with nested properties without having to manually define the class schema.
```

Comment:
Is auto-schema a feature like indexing? In the other doc is is capitalized Auto-schema as if it is more of a product. I think auto-schema is better, but I don't know the product view.

Reply:
it's a feature yup.